src/testdir/test_edit.vim

```diff
@@ -1324,30 +1324,40 @@ func! Test_edit_rightleft()
 endfunc
 
 func Test_edit_complete_very_long_name()
-  if !has('unix') || has('mac')
-    " Long directory names only work on Unix.
-    return
-  endif
   " Try to get the Vim window position before setting 'columns'.
   let winposx = getwinposx()
   let winposy = getwinposy()
   let save_columns = &columns
   set columns=2000
-  call assert_equal(2000, &columns)
+  if has('unix')
+    " On windows (Appveyor) '&columns' would be limited to up to 123.
+    call assert_equal(2000, &columns)
+  endif
   set noswapfile
 
   let dirname = getcwd() . "/Xdir"
-  let longdirname = dirname . repeat('/' . repeat('d', 255), 4)
-  let longfilename = longdirname . '/' . repeat('a', 255)
-  call mkdir(longdirname, 'p')
-  call writefile(['Totum', 'Table'], longfilename)
-  new
-  exe "next Xfile " . longfilename
-  exe "normal iT\<C-N>"
+  if has('win32')
+    let path_max = 260
+  else
+    " 'PATH_MAX' on linux is 4096 and on other *nix seems to be 1024.
+    let path_max = 1024
+  endif
```

Comment:
Hmm, I'm concerned maybe it needs more conditions.

```vim
if has('win32')
   let path_max = 260
elseif has('unix')
  let path_max = 1024
elseif has(...)
  let path_max = ...
...
else
  " skip this test
  let path_max = -1
endif
```

Or maybe it is enough win32 and others. What do you think?